include/fastdds/dds/subscriber/qos/DataReaderQos.hpp

```diff
@@ -37,84 +137,690 @@ namespace dds {
+            const OwnershipQosPolicy& new_value)
+    {
+        ownership_ = new_value;
+    }
+
+    /**
+     * Getter for TimeBasedFilterQosPolicy
+     * @return TimeBasedFilterQosPolicy reference
+     */
+    TimeBasedFilterQosPolicy& time_based_filter()
+    {
+        return time_based_filter_;
+    }
+
+    /**
+     * Getter for TimeBasedFilterQosPolicy
+     * @return TimeBasedFilterQosPolicy const reference
+     */
+    const TimeBasedFilterQosPolicy& time_based_filter() const
+    {
+        return time_based_filter_;
+    }
+
+    /**
+     * Setter for TimeBasedFilterQosPolicy
+     * @param new_value new value for the TimeBasedFilterQosPolicy
+     */
+    void time_based_filter(
+            const TimeBasedFilterQosPolicy& new_value)
+    {
+        time_based_filter_ = new_value;
+    }
+
+    /**
+     * Getter for ReaderDataLifecycleQosPolicy
+     * @return ReaderDataLifecycleQosPolicy reference
+     */
+    ReaderDataLifecycleQosPolicy& reader_data_lifecycle()
+    {
+        return reader_data_lifecycle_;
+    }
+
+    /**
+     * Getter for ReaderDataLifecycleQosPolicy
+     * @return ReaderDataLifecycleQosPolicy const reference
+     */
+    const ReaderDataLifecycleQosPolicy& reader_data_lifecycle() const
+    {
+        return reader_data_lifecycle_;
+    }
+
+    /**
+     * Setter for ReaderDataLifecycleQosPolicy
+     * @param new_value new value for the ReaderDataLifecycleQosPolicy
+     */
+    void reader_data_lifecycle(
+            const ReaderDataLifecycleQosPolicy& new_value)
+    {
+        reader_data_lifecycle_ = new_value;
+    }
+
+    /**
+     * Getter for LifespanQosPolicy
+     * @return LifespanQosPolicy reference
+     */
+    LifespanQosPolicy& lifespan()
+    {
+        return lifespan_;
+    }
+
+    /**
+     * Getter for LifespanQosPolicy
+     * @return LifespanQosPolicy const reference
+     */
+    const LifespanQosPolicy& lifespan() const
+    {
+        return lifespan_;
+    }
+
+    /**
+     * Setter for LifespanQosPolicy
+     * @param new_value new value for the LifespanQosPolicy
+     */
+    void lifespan(
+            const LifespanQosPolicy& new_value)
+    {
+        lifespan_ = new_value;
+    }
+
+    /**
+     * Getter for TopicDataQosPolicy
+     * @return TopicDataQosPolicy reference
+     */
+    TopicDataQosPolicy& topicData()
+    {
+        return topic_data_;
+    }
+
+    /**
+     * Getter for TopicDataQosPolicy
+     * @return TopicDataQosPolicy const reference
+     */
+    const TopicDataQosPolicy& topicData() const
+    {
+        return topic_data_;
+    }
+
+    /**
+     * Setter for TopicDataQosPolicy
+     * @param new_value new value for the TopicDataQosPolicy
+     */
+    void topicData(
+            const TopicDataQosPolicy& new_value)
+    {
+        topic_data_ = new_value;
+    }
+
+    /**
+     * Getter for DurabilityServiceQosPolicy
+     * @return DurabilityServiceQosPolicy reference
+     */
+    DurabilityServiceQosPolicy& durabilityService()
```

Comment:
```suggestion
    DurabilityServiceQosPolicy& durability_service()
```
The same for the other getter and setter